dj_cqrs/transport/rabbit_mq.py

```diff
@@ -57,24 +58,32 @@ def consume(cls):
 
     @classmethod
     def produce(cls, payload):
-        # TODO: try to produce and reconnect several times, now leave as before
-        # if cannot publish message - drop it and try to reconnect on next event
-        rmq_settings = cls._get_common_settings()
-        exchange = rmq_settings[-1]
-
         try:
-            # Decided not to create context-manager to stay within the class
-            _, channel = cls._get_producer_rmq_objects(*rmq_settings)
-
-            cls._produce_message(channel, exchange, payload)
-            cls.log_produced(payload)
+            cls._produce(payload)
         except (exceptions.AMQPError, exceptions.ChannelError, exceptions.ReentrancyError):
-            logger.error("CQRS couldn't be published: pk = {} ({}).".format(
+            logger.error("CQRS couldn't be published: pk = {} ({}). Reconnect...".format(
                 payload.pk, payload.cqrs_id,
             ))
 
             # in case of any error - close connection and try to reconnect
             cls.clean_connection()
+            # reconnect at least 1 time
+            try:
+                cls._produce(payload)
+            except (exceptions.AMQPError, exceptions.ChannelError, exceptions.ReentrancyError):
```

Comment:
Think it's better to make a cycle for DRY, but what's more important is that `cls.clean_connection()` will be called in  any case.